src/test/java/temp/tests/TestGuestCheckout.java

```diff
@@ -0,0 +1,90 @@
+package temp.tests;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import pages.BasePage;
+import pages.LoginPage;
+
+import java.time.Duration;
+
+import static engine.drivers.WebDriverFactory.quitWebDriver;
+import static engine.property.manager.PropertyManager.PropertyKeys.BASE_URL;
+import static engine.property.manager.PropertyManager.getProperty;
+import static engine.property.manager.PropertyManager.getPropertyManagerInstance;
+
+public class TestGuestCheckout {
+    protected static LoginPage loginPage;
+    protected static BasePage basePage;
+
+    @BeforeClass
+    public static void setUp() {
+        getPropertyManagerInstance();
+        basePage = new BasePage();
+        loginPage = new LoginPage();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        quitWebDriver();
+    }
+
+    @Test
+    public void openPage() {
+        String url = getProperty(BASE_URL);
+        basePage.openUrl(url);
+
+        var welcomeMsg = By.xpath("//div[@class='welcome_msg']");
+        WebDriverWait wait = new WebDriverWait(basePage.driver, Duration.ofSeconds(5));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(welcomeMsg));
+
+        Assert.assertEquals(url, basePage.driver.getCurrentUrl());
+    }
+
+    @Test
+    public void addToCartAndGoToCart() {
+        openPage();
+
+        WebElement addProductToCart = basePage.driver.findElement(By.xpath("//*[@id='block_frame_latest_1770']/div/div[1]/div[2]/div[3]/a"));
+        addProductToCart.click();
+
+        WebElement goToCart = basePage.driver.findElement(By.xpath("/html/body/div/header/div[2]/div/div[3]/ul/li/a"));
+        goToCart.click();
+
+        Assert.assertEquals("Cart URL is not valid", "https://automationteststore.com/index.php?rt=checkout/cart", basePage.driver.getCurrentUrl());
+    }
+
+    @Test
+    public void goToCheckout() {
+        addToCartAndGoToCart();
```

Comment:
nonononononono!!! We cannot make tests that depends one on another:) create class with steps and define them there to reuse them in the tests :) could you also add comment that some of the steps are dummy steps that will be implement later on? Maybe put those into some steps as well so it will be easy to change after the implementation is ready (you can allign with creators if any parameter is required there) :)

Reply:
@Fishgame1 Basic tests rework to be discussed tomorrow - as per my comment above